lib/pbench/server/database/models/active_tokens.py

```diff
@@ -1,28 +1,34 @@
-import datetime
+from datetime import datetime
 from typing import Optional
 
 from sqlalchemy import Column, DateTime, ForeignKey, Integer, String
+from sqlalchemy.orm import relationship
 
 from pbench.server.database.database import Database
 
 
 class ActiveTokens(Database.Base):
-    """Model for storing the active auth tokens associated with a user."""
+    """Model for storing the active auth tokens associated with a user
+
+    Each token is associated with a given User object, and stores its
+    expiration time.
+    """
 
     __tablename__ = "active_tokens"
     id = Column(Integer, primary_key=True, autoincrement=True)
     token = Column(String(500), unique=True, nullable=False, index=True)
-    created = Column(DateTime, nullable=False)
+    expiration = Column(DateTime, nullable=False, index=True)
     user_id = Column(
         Integer,
         ForeignKey("users.id", ondelete="CASCADE"),
         nullable=False,
         # no need to add index=True, all FKs have indexes
     )
+    user = relationship("User", back_populates="auth_tokens")
 
-    def __init__(self, auth_token: str):
+    def __init__(self, auth_token: str, expiration: datetime):
         self.token = auth_token
-        self.created = datetime.datetime.now()
+        self.expiration = expiration
```

Comment:
Huh; I hadn't noticed this before. Normally `declarative_base` classes shouldn't have `__init__` methods; they risk inconsistency, because while they *will* be called on an explicit `ActiveTokens(...)` call, they're not *necessary* (the default is a `kwargs` with keywords matching the column names ... and here the mismatch between `auth_token` and `self.token` is a bit annoying), and they're *not* called when SQLAlchemy creates proxies from the DB... which means the temptation to put extra logic into a constructor is potentially confusing. (You can use validator hooks for that, which will be called.)

That is, the only functional difference if this were removed is that you'd need to be careful to `ActiveTokens(token=value, expiration=value)` instead of `ActiveTokens(auth_token=value, expiration=value)`, which wouldn't work because `auth_token` isn't a column. But I don't like the name mismatch anyway. (And, yeah, it's not new with this PR, but I doubt there are that many constructor calls in the code so cleaning it up probably wouldn't be hard...)

Reply:
I'll fix this.